src/common/init.cpp

```diff
@@ -107,7 +107,7 @@ std::optional<ConfigError> InitConfig(ArgsManager& args, SettingsAbortFn setting
             }
         }
     } catch (const std::exception& e) {
-        return ConfigError{ConfigStatus::FAILED, Untranslated(e.what())};
+        return ConfigError{ConfigStatus::FAILED, Untranslated(std::string{e.what()})};
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/31061#discussion_r1793837788

Opened #31072 to implement suggested simplification. Marking this thread resolved since it it's not meant to block this PR in any case.